src/package-request.js

```diff
@@ -132,21 +132,27 @@ export default class PackageRequest {
     }
   }
 
-  async normalizeRange(pattern: string): Promise<string> {
+  async normalizeRange(name: string, pattern: string): Promise<string> {
```

Comment:
If I remember correctly we can delete every  `Promise.resolve` because `async/await` world we should not use them (i.e. It automatically change the result in Promise.resolve).

Do you think we can start to add some unit tests for package-request functions? 

I am afraid that every function in this file is not try to adress one responsability and we couple the code in function just for code duplication. One another proof of this is that we have many similar names. If we can't name something, maybe there is a reason.